### PR TITLE
Non-unified build fixes, early February 2023 edition

### PR DESCRIPTION
**Pull request by donny-dont**
#### 5e48fabd1efd472a2733d7e493d11db5e9da67a8
<pre>
Non-unified build fixes, early February 2023 edition
<a href="https://bugs.webkit.org/show_bug.cgi?id=251469">https://bugs.webkit.org/show_bug.cgi?id=251469</a>

Reviewed by Yusuke Suzuki.

Fix compile issues. A non-unified build flags additional unused
functions and variables.

* Source/JavaScriptCore/bytecode/BytecodeBasicBlock.cpp:
* Source/JavaScriptCore/bytecode/CallLinkStatus.cpp:
* Source/JavaScriptCore/runtime/TemporalPlainDate.cpp:
* Source/WebCore/Modules/WebGPU/GPUAdapter.cpp:
* Source/WebCore/Modules/fetch/FetchBodyOwner.cpp:
* Source/WebCore/Modules/push-api/PushPermissionState.h:
* Source/WebCore/Modules/push-api/PushSubscription.cpp:
* Source/WebCore/Modules/screen-wake-lock/WakeLockManager.cpp:
* Source/WebCore/Modules/streams/WritableStream.cpp:
* Source/WebCore/Modules/streams/WritableStreamSink.h:
* Source/WebCore/Modules/web-locks/WebLockIdentifier.h:
* Source/WebCore/Modules/web-locks/WebLockManager.cpp:
* Source/WebCore/animation/CSSPropertyAnimation.cpp:
* Source/WebCore/bindings/js/JSServiceWorkerGlobalScopeCustom.cpp:
* Source/WebCore/css/CSSCounterStyleRule.cpp:
* Source/WebCore/css/CSSKeyframeRule.cpp:
* Source/WebCore/css/CSSSupportsRule.cpp:
* Source/WebCore/css/ComputedStyleExtractor.cpp:
* Source/WebCore/css/DOMCSSNamespace.cpp:
* Source/WebCore/css/DOMMatrixReadOnly.cpp:
* Source/WebCore/css/PropertySetCSSStyleDeclaration.cpp:
* Source/WebCore/css/StyleRule.cpp:
* Source/WebCore/css/typedom/DeclaredStylePropertyMap.cpp:
* Source/WebCore/dom/DOMImplementation.cpp:
* Source/WebCore/dom/ElementData.cpp:
* Source/WebCore/editing/FontAttributeChanges.cpp:
* Source/WebCore/editing/FontAttributeChanges.h:
* Source/WebCore/editing/FrameSelection.cpp:
* Source/WebCore/fileapi/BlobURL.cpp:
* Source/WebCore/html/HTMLElement.cpp:
* Source/WebCore/html/HTMLFontElement.cpp:
* Source/WebCore/html/HTMLTablePartElement.cpp:
* Source/WebCore/inspector/agents/InspectorCanvasAgent.h:
* Source/WebCore/loader/DocumentLoader.cpp:
* Source/WebCore/page/EventHandler.cpp:
* Source/WebCore/page/Frame.cpp:
* Source/WebCore/platform/audio/PlatformMediaSession.cpp:
* Source/WebCore/platform/graphics/DisplayRefreshMonitor.cpp:
* Source/WebCore/platform/graphics/displaylists/DisplayListItems.cpp:
* Source/WebCore/platform/graphics/displaylists/DisplayListRecorderImpl.cpp:
* Source/WebCore/rendering/EventRegion.cpp:
* Source/WebCore/rendering/GridLayoutFunctions.cpp:
* Source/WebCore/rendering/RenderObject.cpp:
* Source/WebCore/workers/service/context/ServiceWorkerFetch.cpp:
* Source/WebKit/NetworkProcess/Classifier/ResourceLoadStatisticsDatabaseStore.cpp:
* Source/WebKit/NetworkProcess/NetworkResourceLoader.cpp:
* Source/WebKit/NetworkProcess/SharedWorker/WebSharedWorkerServerToContextConnection.cpp:
* Source/WebKit/UIProcess/WebPageProxy.cpp:
* Source/WebKit/UIProcess/WebProcessPool.cpp:
* Source/WebKit/WebProcess/WebProcess.cpp:

Canonical link: <a href="https://commits.webkit.org/259721@main">https://commits.webkit.org/259721@main</a>
</pre>

https://github.com/WebKit/WebKit/commit/1fd088535b42cc6bf3775b19156f4b7ff99dc4c7

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/6/builds/105696 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/77/builds/14787 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/43/builds/38569 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/8/builds/114921 "Built successfully") | [  ~~🛠 🧪 win~~](https://ews-build.webkit.org/#/builders/10/builds/175061 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/11/builds/109597 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/76/builds/16193 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/85/builds/5984 "Built successfully") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/36/builds/97960 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/12/builds/114727 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/19/builds/111449 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/78/builds/12317 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/3/builds/95299 "Passed tests") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/35/builds/39793 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/9/builds/94181 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/73/builds/26939 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/34/builds/81510 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/1/builds/95421 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/81/builds/8052 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/70/builds/28292 "Passed tests") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/38/builds/93594 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/83/builds/5837 "Built successfully and passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/82/builds/8548 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/84/builds/4878 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/46/builds/30425 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/80/builds/14169 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/62/builds/47839 "Passed tests") | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/37/builds/102308 "Built successfully") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/74/builds/6716 "Built successfully and passed tests") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/79/builds/10101 "Built successfully") | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/45/builds/25513 "Passed tests") | 
| | | | |

### CODE REVIEW

File: Source/WebCore/animation/CSSPropertyAnimation.cpp [hunk 1]
```diff
@@ -340,12 +340,6 @@ static inline RefPtr<ShapeValue> blendFunc(ShapeValue* from, ShapeValue* to, con
     return from->blend(*to, context);
 }
 
-static inline RefPtr<FilterOperation> blendFunc(FilterOperation* from, FilterOperation* to, const CSSPropertyBlendingContext& context, bool blendToPassthrough = false)
-{
-    ASSERT(to);
-    return to->blend(from, context, blendToPassthrough);
-}
-
```
Comment (donny-dont) [inline Source/WebCore/animation/CSSPropertyAnimation.cpp:348]: Not 100% sure this isn't used in a guard

File: Source/JavaScriptCore/runtime/LiteralParser.cpp [hunk 2]
```diff
@@ -165,16 +165,6 @@ ALWAYS_INLINE JSString* LiteralParser<CharType>::makeJSString(VM& vm, typename L
...
-static ALWAYS_INLINE bool cannotBeIdentPartOrEscapeStart(LChar)
-{
-    RELEASE_ASSERT_NOT_REACHED();
-}
-
-static ALWAYS_INLINE bool cannotBeIdentPartOrEscapeStart(UChar)
-{
-    RELEASE_ASSERT_NOT_REACHED();
-}
-
```
Comment (donny-dont) [inline Source/JavaScriptCore/runtime/LiteralParser.cpp]: This was added in https://github.com/WebKit/WebKit/commit/6338e2ba60d88e04ffed6703dfa1028a611fd6dc to remove a warning so not sure if this is going to become an issue

**Review by donny-dont — COMMENTED** [review]
Making some notes. There's issues with the generated files having warnings.

**Review by Constellation — APPROVED** [review]
r=me

**Pull request merged**